core/jvm/src/test/scala/scalaz/zio/IOSpec.scala

```diff
@@ -103,4 +110,89 @@ class IOSpec(implicit ee: org.specs2.concurrent.ExecutionEnv) extends AbstractRT
     unsafeRun(IO.done(terminated)) must throwA(FiberFailure(Unchecked(error)))
     unsafeRun(IO.done(failed)) must throwA(FiberFailure(Checked(error)))
   }
+
+  def testWhen = {
+    var effect: Int              = 0
```

Comment:
Note these `var` can be replaced by `Ref` to keep the test more purely functional.

We're evolving to a style where there's just one `unsafeRun` per test.

Reply:
i.e. you can do:

```scala
unsafeRun(
....
(a must_=== b) and ...
)
```